src/vs/platform/telemetry/common/telemetryService.ts

```diff
@@ -68,12 +70,24 @@ export class TelemetryService implements ITelemetryService {
 
 	private _updateTelemetryLevel(): void {
 		this._telemetryLevel = getTelemetryLevel(this._configurationService);
+		const collectableTelemetry = this._productService.collectableTelemetry;
+		// Also ensure that error telemetry is respecting the product configuration for collectable telemetry
+		if (collectableTelemetry) {
+			this._sendErrorTelemetry = this.sendErrorTelemetry ? collectableTelemetry.error : false;
+			// Make sure the telemetry level from the service is the minimum of the config and product
+			const maxCollectableTelemetryLevel = collectableTelemetry.usage ? TelemetryLevel.USAGE : collectableTelemetry.error ? TelemetryLevel.ERROR : TelemetryLevel.NONE;
```

Comment:
do we ever anticipate having other levels exposed in the product json

Reply:
Not unless the setting adds more levels I don't anticipate this